### ISSUE TITLE
Cannot fully submit options since update 3.1.1

### ISSUE THREAD

**Issue by BertrumUK** [craibo/ha_strava#60]
After the latest update I am unable to go further than the 1st submit option. Any settings on the slider reset the settings as per the picture. 

![image](https://user-images.githubusercontent.com/22644029/206794687-7b0c14d3-736b-4f50-b178-66c79209c969.png)

I tried rolling back and resintalling the add-on but I still get the same issue.

**Comment by BertrumUK** [comment]
With debug enabled I get this error

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 515, in async_unload
    result = await component.async_unload_entry(hass, self)
  File "/config/custom_components/ha_strava/__init__.py", line 748, in async_unload_entry
    remove_listener()
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 635, in <lambda>
    return lambda: self.update_listeners.remove(weak_listener)
ValueError: list.remove(x): x not in list
```

**Comment by craibo** [comment]
Hi @BertrumUK 

I missed a `return` statement in the config flow which I have now resolved and released with 3.1.2.
There is no longer a sensor config screen post the inital configuration screen as it is no longer required. 
Each device is created with all the sensors available.

I'm still in the process of updating the sensors so that they are inline with HA implementation standards and the metrics, change tracking and formatting is more standard and will be far easier to deal with for dashboard displays and templating.

Let me know if you come across any issues I may have missed?

Thanks

**Comment by BertrumUK** [comment]
I was just about to add that I tried the rollback to 3.1.0 then saw the 3.1.2 update - its working now with the new update. Thanks for sorting it out.
